### PR TITLE
Added file on lesson

### PR DESCRIPTION
**Pull request by nsahin**

### CODE REVIEW

File: lessons/MachineLearning/lesson.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+layout: page
+title: Machine Learning
+visible: true
```
Comment (lwjohnst86) [inline lessons/MachineLearning/lesson.md]: Please add below visible:

```
tags:
    - python
    - intermediate
```

File: lessons/MachineLearning/lesson.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+---
+<!-- change visible to true if you want it on the site -->
+
+# My Lesson Title
+
+ - **Authors**: Nil Sahin
+ - **Research field**: Computational Biology and Molecular Genetics
+ - **Lesson topic**: Machine Learning using Python's Sklearn packages
+ - **Lesson content URL**: <link>
```
Comment (lwjohnst86) [inline lessons/MachineLearning/lesson.md]: please add the link: `<https://github.com/UofTCoders/studyGroup/tree/gh-pages/lessons/python/scikit-learn>`

edited by LT to change misc > python

File: lessons/MachineLearning/lesson.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+[Sklearn](http://scikit-learn.org/stable/modules/classes.html#module-sklearn.linear_model) class and function reference page
+
+[Machine Learning course](http://www.cs.toronto.edu/~urtasun/courses/CSC411_Fall16/CSC411_Fall16.html) from Computer Science Department, UOFT
+
+[TPOT](https://rhiever.github.io/tpot/) for optimized machine learning pipelines
+
+[KERAS](https://keras.io/) package for Neural Networks
+
+[Tensorflow Playground](http://playground.tensorflow.org/#activation=tanh&batchSize=10&dataset=circle&regDataset=reg-plane&learningRate=0.03&regularizationRate=0&noise=0&networkShape=4,2&seed=0.69754&showTestData=false&discretize=false&percTrainData=50&x=true&y=true&xTimesY=false&xSquared=false&ySquared=false&cosX=false&sinX=false&cosY=false&sinY=false&collectStats=false&problem=classification&initZero=false&hideText=false) for Deep Neural Networks
```
Comment (lwjohnst86) [inline lessons/MachineLearning/lesson.md]: Could these links please be converted to a list form.

File: lessons/MachineLearning/lesson.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+---
+layout: page
+title: Machine Learning
```
Comment (lwjohnst86) [inline lessons/MachineLearning/lesson.md]: Could you please expand on title to: `"Machine learning with scikit-learn"`

File: lessons/MachineLearning/lesson.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+visible: true
+---
+<!-- change visible to true if you want it on the site -->
+
+# My Lesson Title
```
Comment (lwjohnst86) [inline lessons/MachineLearning/lesson.md]: `My Lesson Title` -> `Machine learning in Python`

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Thanks so much for the PR and lesson addition! There are a few comments to address below. Also, we tend to try to do more code alongs rather than purely a demonstration of a technique(s). As it stands, this is quite a bit of code to try to code along to, even for more advanced users (usually we have more beginner-intermediate attendees as a reference). Is it possible to either focus down on a few pieces of machine learning techniques in python for the code along or split it up so some of it is code along and the remainder is explicitly a demonstration rather than a step by step of the code? If you are hoping to do just a demonstration, could you focus down on the techniques/code shown? It is only an hour and a bit after all.

Thanks again! We really appreciate it :D

**Comment by lwjohnst86** [comment]
Also, could you move all the files from `lessons/MachineLearning/` into `lessons/python/scikit-learn`?

**Comment by nsahin** [comment]
Hi Luke,

I did all of the changes you requested. I know the code is a bit long but many of them are duplicates of the same thing essentially. Would you like me to remove some parts from the notebook? We can also play by the ear and keep whatever part that is not covered to a second tutorial. Please let me know.

Best,
-
Nil Sahin
Graduate Student, Andrews & Morris Labs
Department of Molecular Genetics
Faculty of Medicine, University of Toronto

On Jun 5, 2017, at 1:18 PM, Luke W Johnston <notifications@github.com<mailto:notifications@github.com>> wrote:


Also, could you move all the files from lessons/MachineLearning/ into lessons/python/scikit-learn?

File: lessons/python/scikit-learn/UofTCoders_ML.ipynb [hunk 6]
```diff
@@ -0,0 +1,949 @@
...
+   "metadata": {},
+   "source": [
+    "## Iris dataset\n",
+    "### Samples of 3 different species of Iris flower with 4 features (sepal and petal lengths and widths)\n",
+    "### We will learn:\n",
+    "    -Displaying feature relationships, correlations\n",
+    "    -Dimensionality reduction using PCA\n",
+    "    -Scaling the data\n",
+    "    -Classification and clustering algorithms\n",
+    "    -Compare to another image dataset of handwritten digits"
```
Comment (joelostblom) [inline lessons/python/scikit-learn/UofTCoders_ML.ipynb]: Remove indent for the list to render properly and add spaces after hyphens.

File: lessons/python/scikit-learn/UofTCoders_ML.ipynb [hunk 7]
```diff
@@ -0,0 +1,949 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Boston housing prices dataset\n",
+    "### 506 samples with 13 features with continuous values\n",
+    "### We will learn:\n",
+    "    - Regression\n",
+    "    - Error as a mean of performance"
+   ]
```
Comment (joelostblom) [inline lessons/python/scikit-learn/UofTCoders_ML.ipynb]: Same as for the previous markdown list.

File: lessons/python/scikit-learn/lesson.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+	- intermediate
+---
+<!-- change visible to true if you want it on the site -->
+
+# Machine learning in Python
+
+ - **Authors**: Nil Sahin
+ - **Research field**: Computational Biology and Molecular Genetics
+ - **Lesson topic**: Machine Learning using Python's Sklearn packages
+ - **Lesson content URL**: <https://github.com/UofTCoders/studyGroup/tree/gh-pages/lessons/misc/scikit-learn>
```
Comment (lwjohnst86) [inline lessons/python/scikit-learn/lesson.md:18]: Uff, silly me. Sorry about that, I meant `lessons/python/scikit-learn`. My mistake!

**Review by joelostblom — CHANGES_REQUESTED** [review]
Looks great! I didn't encounter any issues running the notebook code. Thanks Nil!

**Comment by QuLogic** [comment]
We've done a pretty good job of keeping to Python 3 in our lessons, so I'm going to ask that you try to change all the `print` statements to functions. Even the TPOT instructions you've linked recommend Python 3. 

At the last minute, it might be a bit of a pain to update your environment and reinstall everything for Python 3. However, you _don't_ need to update your environment for this change; add `from __future__ import print_function` at the beginning of your notebook and it will continue to work with Python 2.

File: lessons/python/scikit-learn/lesson.md (hunk 8 above)
Comment (QuLogic) [reply lessons/python/scikit-learn/lesson.md:18]: @nsahin FYI, you started from a very old commit here, which is why you don't see the new directory structure. A rebase or merge with the latest `gh-pages` would get things back in sync.

File: lessons/python/scikit-learn/lesson.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
+---
+layout: page
+title: Machine Learning with scikit-learn
+visible: true
+tags:
+	- python
```
Comment (QuLogic) [inline lessons/python/scikit-learn/lesson.md]: Jekyll does not understand the tabs here and doesn't render the lesson; please replace with spaces.

File: lessons/python/scikit-learn/UofTCoders_ML.ipynb [hunk 10]
```diff
@@ -0,0 +1,949 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "%pylab inline\n",
```
Comment (QuLogic) [inline lessons/python/scikit-learn/UofTCoders_ML.ipynb:87]: `%matplotlib inline`

**Comment by nsahin** [comment]
Hi Luke,

Sorry I’m pretty packed by this right now. Did you mean ~/lessons/python/scikit-learn/machine-learning/ or is the one that’s up right now is fine? I don’t understand.

Best,
-
Nil Sahin
Graduate Student, Andrews & Morris Labs
Department of Molecular Genetics
Faculty of Medicine, University of Toronto

**Comment by nsahin** [comment]
Thanks Joel,

I fixed the changes.

Best,
-
Nil Sahin
Graduate Student, Andrews & Morris Labs
Department of Molecular Genetics
Faculty of Medicine, University of Toronto

**Comment by nsahin** [comment]
Hi Elliott,

Can you help me in rebasing or merging with the latest gh-pages so that I have synced folder structure? What commands I should do? I’m pretty new to git.

Best,
-
Nil Sahin
Graduate Student, Andrews & Morris Labs
Department of Molecular Genetics
Faculty of Medicine, University of Toronto

File: lessons/python/scikit-learn/lesson.md (hunk 8 above)
Comment (lwjohnst86) [reply lessons/python/scikit-learn/lesson.md:18]: @nsahin I'll make it clearer. Change line 16, which is right now:

`- **Lesson content URL**: <https://github.com/UofTCoders/studyGroup/tree/gh-pages/lessons/misc/scikit-learn>`

to 

`- **Lesson content URL**: <https://github.com/UofTCoders/studyGroup/tree/gh-pages/lessons/python/scikit-learn>`

The only thing different is the `misc` -> `python` in the link.

**Review by joelostblom — APPROVED** [review]

**Comment by lwjohnst86** [comment]
Thanks @nsahin! Just a tip for next time around/for using git in general, it's usually a good idea (for those looking over your PR/git commit history) to make your commit messages more informative than "Added file on lesson". Just a tidbit of advice :smile:

**Review by lwjohnst86 — APPROVED** [review]

**Comment by nsahin** [comment]
Hi Luke,

I changed the url, I will try to change my commits to be more informative. Thanks for the heads up, pretty new to git as you can assume:)

Best,
-
Nil Sahin
Graduate Student, Andrews & Morris Labs
Department of Molecular Genetics
Faculty of Medicine, University of Toronto

On Jun 6, 2017, at 1:59 PM, Luke W Johnston <notifications@github.com<mailto:notifications@github.com>> wrote:


Merged #275<https://github.com/UofTCoders/studyGroup/pull/275>.

**Comment by lwjohnst86** [comment]
No worries :) I've fixed the url after I merged. I also fixed the `print` to be `print()` as per Python 3 (see [this commit](https://github.com/UofTCoders/studyGroup/commit/fec738ac54f1e9950923907f60ea0700d432529f)). Just FYI.

**Comment by nsahin** [comment]
I was actually on this, thank you so much Luke! Save a lot of time. I feel like I need to actually do the change to switch to python3 for my own codes that I’ve been slacking for a long time.

Is there anything else I should do for the tutorial? I can skip some parts during the tutorial and we can def remove them from the ipynb. Also, someone mentioned if we can show the output of each cell on the notebook, can you also do that?

Best,
-
Nil Sahin
Graduate Student, Andrews & Morris Labs
Department of Molecular Genetics
Faculty of Medicine, University of Toronto

On Jun 6, 2017, at 2:47 PM, Luke W Johnston <notifications@github.com<mailto:notifications@github.com>> wrote:


No worries :) I've fixed the url after I merged. I also fixed the print to be print() as per Python 3 (see this commit<https://github.com/UofTCoders/studyGroup/commit/fec738ac54f1e9950923907f60ea0700d432529f>). Just FYI.

**Pull request merged**